hyperspy/io_plugins/edax.py

```diff
@@ -680,6 +689,11 @@ def spc_reader(filename,
         hyperspy.io.load_with_reader
     """
     with open(filename, 'rb') as f:
+        # Get version information from the file, and then seek to the beginning
+        version = np.fromfile(f, dtype=[('version',
+                                         '{}f4'.format(endianess))])
+        f.seek(0)
+
         spc_header = np.fromfile(f,
```

Comment:
This is an option, but regardless the data is read the same way, so there shouldn't be any case (that I can think of) where one would fail and the other would succeed. I can code it up that way if desired, though.

Reply:
Since the original issue was related to errors when reading some metadata, I would assume that reading a minimum set of metadata as fallback would still allow to read the file. Was it possible to read the file with `load_all=False` or not at all? If not, forget my comment.

Reply:
Ultimately, the issue was that the new file versions for .ipr and .spd files are slightly larger than the old ones. Since I originally coded this up only with access to the new files, the `np.fromfile` command would return empty on the old files, since I was trying to read "too much" information from the file with the `get_XXX_dtype_list` methods. This then caused the `TraitsError` further down the chain. 

Since it appears that EDAX maintains backwards compatibility in their file structure, it might be smarter to read the file one value at a time by stepping through the dtype list (i.e. with many calls to `np.fromfile`), rather than reading everything from the file in one fell swoop with a giant dtype list. That way if the file read fails, at least we'd have the previously-read values. 

Although since they only change the file structure once every few years, I'm not sure it's worth the effort to re-code the reader. As is, a new file version that has new information at the end of the file will still work, the new values just won't be read. The file versions in #1771 that were causing issues were from a circa 2004 software version, so I'd be very surprised if we ran into any older versions than those.

Either way, the modified version should work on the older (and newest) file versions, and presumably on the ones in-between, but I have no examples of those files to test on.